docs/src/description/declarations.md

```diff
@@ -0,0 +1,74 @@
+# Type declaration sections
+
+## trace_columns
+
+A `trace_columns` section contains declarations for `main` trace columns or `aux` (auxiliary) trace columns.
+
+The `main` and `aux` declarations define the shape of the main and auxiliary execution traces respectively and declare identifiers which can be used to refer to each of the columns in that trace.
+
+**A `trace_columns` section with a `main` declaration is required for an Air DSL to be valid.** The `aux` declaration is optional.
+
+The following is a valid `trace_columns` block:
+
+```
+trace_columns:
+    main: [a, b, c]
+    aux: [d, e]
+```
+
+In the above example, the main execution trace for the Air has 3 columns which can be referenced by `a`, `b`, and `c`. Internally, these identifiers will reference the trace columns with indices 0, 1, and 2 respectively. Similarly, the auxiliary execution trace has 2 columns which can be referenced by `d` and `e`.
+
+## public_inputs
+
+A `public_inputs` section contains declarations for public inputs. Currently, each public input must be provided as a vector of a fixed size, but there is no limit to how many of them can be declared within the `public_inputs` section.
+
+**Public inputs are required.** There must be at least one pubic input declared.
+
+Each public input is described by an identifier and an array length `n` in the following format:
```

Comment:
nit: probably should be "array of length `n`" (with "of").

Reply:
Actually, I intended it as "Each public input is described by an identifier and an array length." I'll put `n` in parentheses.